thoth/prescriptions_refresh/handlers/image_analysis.py

```diff
@@ -0,0 +1,218 @@
+
+    for package in pipfile.packages.packages:
+        if package in _REPRESENTATIVE_PACKAGES:
+            version = pipfile_lock.packages[package].version
+            required_packages.append({"name": package, "version": version})  # Includes ==
+
+    line = 0
+    for package in required_packages:
+        if line == 0:
+            resolved_dependencies += f"- name: {package['name']}\n"
+        else:
+            resolved_dependencies += f"        - name: {package['name']}\n"
+
+        resolved_dependencies += f"          version: {package['version']}\n"
+        line += 1
+
+    return resolved_dependencies
+
+
+def thoth_image_analysis(prescriptions: "Prescriptions") -> None:
+    """Create prescriptions from Thoth container image analysis results."""
+    if not QUAY_TOKEN:
+        raise ValueError("No Token to Quay API provided")
+
+    for image in sorted(chain(get_ps_s2i_image_names(), get_configured_image_names())):
+
+        # Get tag for Thoth images hosted on Quay
+        for _, tag in get_image_containers(image):
+            _LOGGER.info("Obtaining image information for %r in tag %r", image, tag)
+
+            image_url = f"{QUAY_URL}/thoth-station/{image}:{tag}"
+
+            # Get image analyzed IDs latest from database through USER-API endpoint
+            info = _get_latest_image_analyzed_info(image=image)
+
+            if not info:
+                _LOGGER.warning("Could not find any data for {QUAY_URL}/thoth-station/{image}:{tag} in Thoth Database.")
+                break
+
+            # Software environment
+            os_name = info["os_name"]
+            os_version = info["os_version"]
+            python_version = info["python_version"]
+
+            # Latest package extract document ID
+            package_extract_document_id = info["package_extract_document_id"]
+
+            # Get Pipfile/Pipfile.lock from image analyzed result
+            pipfile_dict, pipfile_lock_dict = _get_requirement_files_from_image_analysis(
+                package_extract_document_id=package_extract_document_id
+            )
+
+            if pipfile_dict and pipfile_lock_dict:
+                # Create section for locked packages versions for prescriptions
+                resolved_dependencies = _create_resolved_dependencies_section(pipfile_dict, pipfile_lock_dict)
+
+                if not resolved_dependencies:
+                    _LOGGER.error(
+                      "No representative packages identified, please check list of packages stated."
+                    )
+                    break
+
+                # Create prescriptions for direct dependencies
+                prescriptions.create_prescription(
+                    project_name="_containers",
+                    prescription_name=_IMAGE_ANAYSIS_PRESCRIPTION_NAME,
+                    content=_THOTH_IMAGE_ANALYSIS_WRAP.format(
+                        prescription_name=Prescriptions.get_prescription_name("BaseImageWrap", f"Thoth-{image}-{tag}"),
```

Comment:
Prescription name is alphanum:

```suggestion
                        prescription_name=Prescriptions.get_prescription_name("BaseImageWrap", "Thoth", image, tag),
```